components/form/index.js

```diff
@@ -199,11 +217,29 @@ const ApplicationForm = () => {
                 {...{
                   instanceId: 'resources',
                   name: 'resources',
-                  onChange: setFieldValue,
+                  onChange: (e, value) => {
+                    if (value && value.find( (res) => res.value === 'other') ) {
+                      setHideOther(false)
+                    } else {
+                      setHideOther(true)
+                    }
+                    setFieldValue(e,value)
+                  },
                   onBlur: setFieldTouched,
                   options: SelectOptions,
                 }}
               />
+              {!hideOther && errors.other && touched.other ? (
+                <ErrorMessage message={errors.other} />
+              ) : null}
+              <Input
+                {...{
+                  id: 'other',
+                  name: 'other',
+                  placeholder: 'Please describe the other resource(s)',
+                  type: hideOther ? 'hidden' : 'text',
+                }}
```

Comment:
is there a reason why we use `{...{prop: 'prop'}}` instead of normal prop syntax?

Reply:
@robyn3choi No, was just done some way in some files and it turns out that way -- use either or/whatever makes sense in that context. Spreading helps when there are lots of duplicate values. For example:

```
a={a}
b={b}
c={c}
```
vs
```
{...{a, b, c}} // Works because of object property shorthand notation in ES2015, you can "leave out the right side basically"
```

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#new_notations_in_ecmascript_2015

Reply:
Another benefit is you can stay in JS instead of JSX (which can also be seen as a con). For this project, my opinion is to use it sparingly because many people coming on/off this project are more used to regular JSX without fancy spread/object prop shorthand notations.

Reply:
It looks like the spread notation for props is only used in this file (and once in `components/button/index.js`). I've never seen props like this before and spent a few min figuring out what was going on, so I agree that we should switch it to the normal prop notation. Especially since there aren't duplicate values here like in your example above.

Reply:
The reason it was still in `form/index.js` is that it used the shorthand notation for the `onChange` function prior to adding the Formik validation. Formik handles the onChange now, and it was removed with that PR. I'll switch it back to traditional prop notation as it makes it easier to read going forward, especially if others will join in.